7-React/src/components/TodoItem/TodoItem.driver.tsx

```diff
@@ -0,0 +1,101 @@
+import TodoItem from "./TodoItem";
+import { fireEvent, render, RenderResult } from "@testing-library/react";
+import dataHooks from "./data-hooks";
+import React from "react";
+
+class TodoItemDriver {
+  private wrapper?: RenderResult;
+  private todoDescription: string = "";
+  private deleteTaskMock: jest.Mock = jest.fn();
+  private editTextMock: jest.Mock = jest.fn();
+  private toggleDone: jest.Mock = jest.fn();
+
+  private editBtn = () => {
+    const element = this.wrapper?.queryByTestId(dataHooks.editBtn);
+    return {
+      editBtnExists: () => {
+        return element ? true : false;
+      },
+      editBtnClick: () => {
+        if (element) {
+          fireEvent.click(element);
+        }
+      },
+    };
+  };
+  private deleteBtn = () => {
+    const element = this.wrapper?.queryByTestId(dataHooks.deleteBtn);
+    return {
+      deleteBtnExists: () => {
+        return element ? true : false;
+      },
+      deleteTaskClick: () => {
+        if (element) {
+          fireEvent.click(element);
+        }
+      },
+    };
+  };
+  private taskTitle = () => {
+    const element = this.wrapper?.queryByTestId(dataHooks.title);
+    return {
+      getTitle: () => {
+        return element?.textContent;
+      },
+    };
+  };
+  private checkBoxBtn = () => {
+    const element = this.wrapper?.queryByTestId(dataHooks.checkbox);
+    const clickCheckBox = () => {
+      if (element) {
+        fireEvent.click(element);
+      }
+      
+    }
+    return {
+      checkBoxBtnExists: () => {
+        return element ? true : false;
+      },
+      markAsDone: () => {
+        clickCheckBox()
+      },
+      unMarkAsDone: () => {
+        clickCheckBox();
+      },
```

Comment:
you're assuming someone will call `markAsDone` first? either check the status before clicking (so that `markAsDone` will only succeed if it actually marked an item as done).
another option is to unify them into `toggleDone` or similar (your call)